source/features/highlight-deleted-and-added-files-in-diffs.tsx

```diff
@@ -0,0 +1,69 @@
+import select from 'select-dom';
+import oneTime from 'onetime';
+import {observe} from 'selector-observer';
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import {observeOneMutation} from '../helpers/simplified-element-observer';
+
+async function loadDeferred(jumpList: Element): Promise<void> {
+	const loadJumpList = (jumpList: Element) => jumpList.parentElement!.dispatchEvent(new MouseEvent('mouseover'));
+	loadJumpList(jumpList);
+	// The event listener might not have been attached yet, so we can try twice
+	setTimeout(loadJumpList, 1000, jumpList);
+	await observeOneMutation(jumpList);
+}
+
+async function init(): Promise<void> {
+	const fileList = await elementReady([
+		'.toc-select details-menu', // `isPr`
+		'.toc-diff-stats + .content' // `isRepoCommit`
+	].join());
+	if (pageDetect.isPR()) {
+		await loadDeferred(fileList!);
+	}
+
+	observe('.file-info [href]:not(.rgh-pr-file-state)', {
+		constructor: HTMLAnchorElement,
+		add(element) {
+			element.classList.add('rgh-pr-file-state');
+			let icon = select([
+				`[href="${element.hash}"] svg`, // `isPr`
+				`svg + [href="${element.hash}"]` // `isRepoCommit`
+			], fileList)!;
+			// `isRepoCommit` needs previousElementSibling
+			icon = (icon.previousElementSibling as HTMLElement ?? icon).cloneNode(true);
```

Comment:
```suggestion
			const icon: HTMLElement =
				select(`[href="${element.hash}"] svg`, fileList) ??  // `isPr`
				select(`svg + [href="${element.hash}"]`, fileList)?.previousElementSibling!; // `isRepoCommit`
			).cloneNode(true);
```

or if that doesn't work

```suggestion
			const icon =
				select(`[href="${element.hash}"] svg`, fileList) ??  // `isPr`
				select(`svg + [href="${element.hash}"]`, fileList)?.previousElementSibling as HTMLElement; // `isRepoCommit`
			).cloneNode(true);
```